rfc: add new pipeline

yorkie opened the pull request:
See the markdown to review.

yorkie commented:
I will add JSON schema later, please have reviews on other parts.

yorkie commented:
Added JSON schema definition, @WenheLI @rickycao-qy.

yorkie commented:
@rickycao-qy May I ask you to update the RL parts on this proposal?

rickycao-qy commented on docs/rfcs/0000-new-pipeline.md:
```diff
@@ -0,0 +1,252 @@
+      "package": "@pipcook/plugins-tfjs-text-lstm-model-evaluate"
+    },
+    {
+      "type": "deploy",
```
Is it better to specify the type of 'modelDeploy', just to be in conformity with other plugins' types

yorkie replied:
The name "modelDeploy" is the deprecated type, I use the new and common name is that what the plugin to deploy is not only the model, includes the runtime, data processing, and others.

rickycao-qy commented on docs/rfcs/0000-new-pipeline.md:
````diff
@@ -0,0 +1,252 @@
+```json
+{
+  "specVersion": "2.0",
+  "plugins": [
````
what if the user specifies more than one dataCollect/dataAccess/modelDeinfe plugins? Shall we make the whole plugins to be an array or just dataProcess to be an array?

yorkie replied:
To make sure the extensibility of plugins, I think we could keep all plugins should be in the array, how about add additional rules to restrict the array such as limit only 1 "modelDefine" type?

rickycao-qy replied:
This sounds good. We can modify the schema so that the numbers are limited to 1 for now.

yorkie replied:
@rickycao-qy I have updated the schema below, the plugin items are split into multiple sub-schemas, which I defined the `maxItems` and `minItems` in different ones.

yorkie commented:
Within the separation of plugin items, I also defined some of the common parameters for every specific type of plugin, for example,

- added `url` to the params once the plugin type is "dataCollect"
- added `epochs` to the params once the plugin type is "modelTrain"
- ...

And we need to make sure these names are consistent with our core interfaces for plugin developers.

rickycao-qy commented:
> Within the separation of plugin items, I also defined some of the common parameters for every specific type of plugin, for example,
> 
> * added `url` to the params once the plugin type is "dataCollect"
> * added `epochs` to the params once the plugin type is "modelTrain"
> * ...
> 
> And we need to make sure these names are consistent with our core interfaces for plugin developers.

Just wondering if it's necessary to add these common parameters cuz the user may implement specific data-collect plugin just to collect specific data without url.

yorkie commented:
How about making the URL in `data-collect` be optional?

rickycao-qy commented:
> How about making the URL in `data-collect` be optional?

Sounds good to me

yorkie commented:
> > How about making the URL in `data-collect` be optional?
> 
> Sounds good to me

Updated :)

rickycao-qy reviewed (approved):
LGTM

FeelyChau reviewed (approved):
LGTM.

WenheLI reviewed (approved):
LGTM
